pkg/controllers/namespacesync/references.go

```diff
@@ -60,13 +60,17 @@ func walkReferences(
 	for _, ref := range []*corev1.ObjectReference{
 		cc.Spec.Infrastructure.Ref,
 		cc.Spec.ControlPlane.Ref,
-		cc.Spec.ControlPlane.MachineInfrastructure.Ref,
 	} {
 		if err := fn(ctx, ref); err != nil {
 			return err
 		}
 	}
-
+	// Managed kubernetes providers like EKS and AKS will not have a MachineInfrastructure reference for control plane.
```

Comment:
I think `walkReferences` is not responsible for knowing when references might be nil. Instead, let's please handle nil references in a generic way. I think this bug is also present for workers, in the loop below.

Reply:
added nil checks for all references. https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1366/commits/69ac5fa9ce0a9856e1b6bdba439ce81e459e48ff